Asterix/Main_EFC_THD.py

```diff
@@ -602,36 +654,34 @@ def correctionLoop(parameter_file,
     if set_amplitude_abb == True:
         if ampl_abb_filename != '' and os.path.isfile(
                 Model_local_dir + ampl_abb_filename +
-                ".fits") == True and set_random_ampl == False:
+                ".fits") == True and set_random_ampl is False:
             ampfinal = phase_ampl.scale_amplitude_abb(
                 model_dir + ampl_abb_filename + ".fits", thd2.prad,
                 thd2.entrancepupil.pup)
         else:
             ampl_abb_filename = "ampl_{:d}rms_spd{:d}_rhoc{:.1f}_rad{:d}".format(
                 int(ampl_rms), int(ampl_slope), ampl_rhoc, thd2.prad)
 
-            if set_random_ampl == False and os.path.isfile(Model_local_dir +
+            if set_random_ampl is False and os.path.isfile(Model_local_dir +
                                                            ampl_abb_filename +
                                                            ".fits") == True:
                 ampfinal = fits.getdata(Model_local_dir + ampl_abb_filename +
-                                    ".fits")
+                                        ".fits")
             else:
                 ampfinal = thd2.entrancepupil.random_phase_map(
                     ampl_rms / 100., ampl_rhoc, ampl_slope)
-            if set_random_ampl == False:  # save it for next time
+            if set_random_ampl is False:  # save it for next time
                 fits.writeto(Model_local_dir + ampl_abb_filename + ".fits",
-                             ampfinal)
+                             ampfinal, overwrite=True)
     else:
         ampfinal = 0
 
     amplitude_abb_up = ampfinal
     phase_abb_up = phase_up
 
     ## To convert in photon flux
-    # We can probably replace here by transmission !
-    contrast_to_photons = (np.sum(thd2.entrancepupil.pup) /
-                           np.sum(thd2.corono.lyot_pup.pup) * nb_photons *
-                           thd2.maxPSF / thd2.sumPSF)
+    contrast_to_photons = 1 / thd2.transmission() * nb_photons * thd2.maxPSF / thd2.sumPSF
```

Comment:
Why maxPSF/sumPSF and why 1/transmission and not transmission?
transmission  = sum(pupil out) / sum(pupil in)
nb photon out = nb photons * transmission

Not sure what contrast_to_photons represents.

Reply:
this has been changed